src/mapping.ts

```diff
@@ -10,18 +10,36 @@ import { tokens, trades, orders, users } from "./modules"
 export function handleInteraction(event: Interaction): void {}
 
 export function handleOrderInvalidated(event: OrderInvalidated): void {
+
   let orderId = event.params.orderUid.toHexString()
-  orders.invalidateOrder(orderId)
+  let timestamp = event.block.timestamp
+
+  let order = orders.invalidateOrder(orderId, timestamp)
+
+  order.save()
 }
 
-export function handlePreSignature(event: PreSignature): void {}
+export function handlePreSignature(event: PreSignature): void {
+
+  let orderUid = event.params.orderUid.toHexString()
+  let ownerAddress = event.params.owner
+  let owner = ownerAddress.toHexString()
+  let timestamp = event.block.timestamp
+
+  let order = orders.preSig(orderUid, owner, timestamp)
+
+  order.save()
+
+  users.getOrCreateUser(owner, timestamp, ownerAddress)
+}
 
 export function handleSettlement(event: Settlement): void {}
 
 
 export function handleTrade(event: Trade): void {
   let orderId = event.params.orderUid.toHexString()
-  let owner = event.params.owner
+  let ownerAddress = event.params.owner
+  let owner = ownerAddress.toHexString()
```

Comment:
same comment, we need to be thorough with the params. You are not including the amounts. I understand they will belong as both part of the trade, and part of the token (we accumulate all the amounts)